homeworks/09-node-chokidar/src/dirWatcher/index.ts

```diff
@@ -0,0 +1,42 @@
+import { EventEmitter } from "events"
+import {EventName, FileName} from "../domain"
+import {Set} from "immutable"
+import path from "path"
+import * as fs from "fs"
+import {CancellationFunction, delay} from "../util"
+
+export class DirWatcher {
+    constructor(
+        private readonly eventEmitter: EventEmitter,
+    ) {}
+
+    public watch(directoryPath: string, interval = 1000): CancellationFunction {
+        let processed: Set<FileName> = Set()
+        let keepRunning = true
+
+        const process = (file: FileName): void => {
+            if (!processed.contains(file)) {
```

Comment:
I think there should be a file filter here otherwise I can drop any file type and it will output in console (e.g. image).
The task describes processing only CSV files.